Assets/Scripts/Editor/AgentTypeAttributeDrawer.cs

```diff
@@ -0,0 +1,55 @@
+﻿using UnityEditor;
+using UnityEditor.AI;
+using UnityEngine;
+using UnityEngine.AI;
+
+namespace Bug.Navigation
+{
+	[CustomPropertyDrawer(typeof(AgentTypeAttribute))]
+	public class AgentTypeAttributeDrawer : PropertyDrawer
+	{
+		public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
+		{
+			int index = -1;
+			int count = NavMesh.GetSettingsCount();
+			string[] agentTypeNames = new string[count + 2];
```

Comment:
You can just write = new[count + 2]; here

Reply:
Thank you for the tip! I think I prefer it that way.